### ISSUE TITLE
[META] Automate Integration testing

### ISSUE THREAD

**Issue by setiah** [opensearch-project/opensearch-build#124]
**This is a meta issue for tracking all sub-tasks for full integration testing on OpenSearch distribution.**

P0
- [x] #178
- [x] #182
- [x] #183
- [x] #327
- [x] #382
- [x] https://github.com/opensearch-project/opensearch-build/issues/395
- [x] [**BLOCKER**] #319 

P1

- [x] #220
- [x] #473 
- [x] https://github.com/opensearch-project/opensearch-build/issues/1113
- [x] #214
- [ ] #326
- [ ] #202
- [ ] Parallelize integration testing for plugins

Depends on #122 

- [ ] https://github.com/opensearch-project/opensearch-build/issues/704
- [ ] https://github.com/opensearch-project/opensearch-build/issues/818
- [ ] https://github.com/opensearch-project/opensearch-build/issues/58

**Comment by setiah** [comment]
**Describing integration test job in its current form**

1. Clone opensearch (1.0), opensearch-build and plugin (based on ref provided) to test. Then clone the dependencies - common-utils, job-scheduler, alerting as notifications.
2. Deploy single node cluster based on params VERSION, TYPE (snapshot | release), ARCHITECTURE (x64 | amd64), SECURITY (w/ | w/o). This downloads artifacts from https://artifacts.opensearch.org, adds settings, disables security (if required), and starts OpenSearch process. It validates the process is running by doing a curl on the ports.
3. Next it pulls integ test dependencies using [integtest_dependencies_opensearch.sh](https://github.com/opensearch-project/opensearch-build/blob/main/tools/standard-test/integtest_dependencies_opensearch.sh) into maven-local, based on VERSION (e.g. 1.0.0) and QUALIFIER (e.g. rc1, ga) for each of the repos pulled in step 1 (opensearch, common-utils, job-scheduler, alerting). It uses `./gradlew publishToMavenLocal` to publish dependencies to maven local. These dependencies should now be available in `~/.m2/repository/org/opensearch`
4. Next it moves the plugin repo to `opensearch/plugins` folder and executes the `integtest.sh` from that plugin repo against provided CLUSTER_PORT

**Comment by setiah** [comment]
Inputs required:
1. Artifact download path
2. git-reference (commit id, etc.)
3. Architecture
4. Security
5. Log publish path (based on `BuildId`)

What the job needs to do
1. Add parallelization logic to run tests in parallel on plugins. 
2. Spin up multiple single node clusters - one for each parallel test run
3. Publish logs for all runs under common build id, with unique test id.

**Comment by peternied** [comment]
This item is redundant with https://github.com/opensearch-project/opensearch-build/issues/818 closing this one in favor of that one

**Comment by dblock** [comment]
Going to reopen this until we run integration tests with every build.

**Comment by dblock** [comment]
Closing in favor of https://github.com/opensearch-project/opensearch-build/issues/818
